### PR TITLE
Solve metadata issues.

### PR DESCRIPTION
**Pull request by jason-fox**
Bug Fix related to Question on [StackOverflow](https://stackoverflow.com/questions/62001030/not-supporting-ngsiv2-metadata-in-device-provisioned-attributes). Additional work is related to fixing bugs found with #836 

The templates checking a valid provisioning request did not accept the `metadata` attrribute - `metadata` could only be defined in the config. Whilst tracking this down I discovered that the Timestamp formatter plugin was overriding values of non-DateTime metadata whenever measures were being received.

- Add templates
- Reduce scope of metadata plugins
- Add tests

### CODE REVIEW

File: lib/plugins/pluginUtils.js [hunk 1]
```diff
@@ -114,7 +114,9 @@ function createProcessAttribute(fn, attributeType) {
             } else if (attribute.metadata) {
                 Object.keys(attribute.metadata).forEach(function(key) {
                     var entry = attribute.metadata[key];
-                    entry.value = fn(entry.value);
+                    if(entry.type === 'DateTime'){
```
Comment (fgalan) [inline lib/plugins/pluginUtils.js:117]: Entry in CHANGE_NEXT_RELEASE about this fix should be added (mentioning issue #869), pls.

Comment (jason-fox) [reply lib/plugins/pluginUtils.js:117]: Fixed 4d54070

**Comment by onrao** [comment]
@fgalan  please can we get this patch merged to the mainline

**Comment by achandana** [comment]
Can some one please review the code ande approve it.

**Comment by fgalan** [comment]
There is a small conflict in CNR file. Once it gets solved, this PR will be merged.

**Comment by achandana** [comment]
Thanks @jason-fox  and @fgalan. Fix is working fine for us.

**Pull request merged**